### PR TITLE
use OpenTelemetry recommended method to record RPC errors

### PR DESCRIPTION
**Pull request by paulius-valiunas**

### CODE REVIEW

File: core/bentley/src/Tracing.ts [hunk 1]
```diff
@@ -22,6 +23,8 @@ export enum SpanKind {
   CONSUMER = 4
 }
 
+/* eslint-disable deprecation/deprecation */
```
Comment (pmconne) [inline core/bentley/src/Tracing.ts:26]: Please use `eslint-disable-line`/`next-line`.

File: core/bentley/src/Tracing.ts [hunk 2]
```diff
@@ -82,6 +85,7 @@ function flattenObject(obj: object): SpanAttributes {
 /**
  * Enables OpenTelemetry tracing in addition to traditional logging.
  * @public
+ * @deprecated in 4.4 - OpenTelemetry Tracing helpers will become internal in a future release. Apps should use `@opentelemetry/api` directly.
```
Comment (pmconne) [inline core/bentley/src/Tracing.ts:88]: Document deprecations in NextVersion.md.

File: core/bentley/src/Tracing.ts (hunk 1 above)
Comment (paulius-valiunas) [reply core/bentley/src/Tracing.ts:26]: the Tracing class has 29 references to itself, can we make an exception for this file? 😅 I moved it closer to the class to reduce collateral damage, let me know if that's ok for you

File: core/bentley/src/Tracing.ts (hunk 2 above)
Comment (paulius-valiunas) [reply core/bentley/src/Tracing.ts:88]: done

**Review by pmconne — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Comment by paulius-valiunas** [comment]
something wrong with iOS tests again? I can't find the actual error

**Pull request merged**